docs/develop/publiccapi.rst

```diff
@@ -0,0 +1,21 @@
+.. _publiccapi:
+
+The public C API
+=================
+
+The public C-API comprises functions defined in all ``C`` header files in the ``py``
+directory. Most of the important core runtime C APIS are exposed in ``runtime.h`` and
+``obj.h``.
+
+An example of such an API is one for LED manipulations exposed in the ``nrf`` port:
```

Comment:
This is not a public API.  (In fact, probably best not to use anything from the nRF port as examples of best practices as there are a few differences to other ports).

Maybe a simple example would be

```
mp_obj_t mp_obj_new_tuple(size_t n, const mp_obj_t *items);
void mp_obj_tuple_get(mp_obj_t self_in, size_t *len, mp_obj_t **items);
void mp_obj_tuple_del(mp_obj_t self_in);
mp_int_t mp_obj_tuple_hash(mp_obj_t self_in);
```